src/hash.cr

```diff
@@ -438,6 +446,15 @@ class Hash(K, V)
     array
   end
 
+  # Appends *other* to a `Hash`, but does not alter it.
```

Comment:
I'm not sure append is the right word. And not just any `Hash`, but the receiver.

> Returns a new hash with the keys and values of this hash and \*other\* combined. A value in *other* takes precedence over the one in this hash.

Something like this? Similar below.